### PR TITLE
feat(core): add CSS function notation syntax for "stMap" and "stArray"

### PR DESCRIPTION
**Pull request by tzachbon**
- [x] - remove `processDeclarationUrls` and use `processDeclarationFunctions` 
- [x] - add `st-array`, `st-map`
- [x] - show deprecation warning
- [x] - update documentation https://github.com/wixplosives/stylable.io/pull/28

### CODE REVIEW

File: packages/core/src/stylable-processor.ts [hunk 1]
```diff
@@ -1055,3 +1088,33 @@ export interface DeclStylableProps {
 export interface SDecl extends postcss.Declaration {
     stylable: DeclStylableProps;
 }
+
+export function processDeclarationFunctions(
+    decl: postcss.Declaration,
+    onFunction: (node: AnyValueNode) => void,
```
Comment (tomrav) [inline packages/core/src/stylable-processor.ts:1094]: why is the type used here (`AnyValueNode`) coming from `css-selector-tokenizer`?

Also, maybe these two utils should go in `stylable-utils.ts`?

File: packages/core/test/functions.spec.ts [hunk 2]
```diff
@@ -678,6 +678,98 @@ describe('Stylable functions (native, formatter and variable)', () => {
     });
 
     describe('diagnostics', () => {
+        describe('deprecation', () => {
+            it('should return a warning when used deprecated stMap', () => {
```
Comment (tomrav) [inline packages/core/test/functions.spec.ts:682]: `should return a warning when using deprecated "stMap" syntax`

File: packages/core/test/functions.spec.ts [hunk 3]
```diff
@@ -678,6 +678,98 @@ describe('Stylable functions (native, formatter and variable)', () => {
...
+                                'st-map'
+                            ),
+                            file: '/style.st.css',
+                            severity: 'info'
+                        },
+                    ]
+                );
+            });
+
+            it('should return a warning when used deprecated stArray', () => {
```
Comment (tomrav) [inline packages/core/test/functions.spec.ts:710]: `should return a warning when using deprecated "st-array" syntax`

File: packages/core/test/functions.spec.ts [hunk 4]
```diff
@@ -678,6 +678,98 @@ describe('Stylable functions (native, formatter and variable)', () => {
...
+                                'st-array'
+                            ),
+                            file: '/style.st.css',
+                            severity: 'info'
+                        },
+                    ]
+                );
+            });
+
+            it('should return a warning when used deprecated nested stArray or stMap', () => {
```
Comment (tomrav) [inline packages/core/test/functions.spec.ts:737]: `should return a warning when using a nested deprecated "stArray" or "stMap" syntax`

**Review by tomrav — APPROVED** [review]
Left a couple of questions/suggestions, looks good otherwise.

**Review by barak007 — APPROVED** [review]

**Pull request merged**